src/XIVLauncher.Core/UnixCompatibility/SteamCompatibilityTool.cs

```diff
@@ -0,0 +1,119 @@
+using System.Numerics;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics;
+using Serilog;
+using XIVLauncher.Core;
+
+namespace XIVLauncher.Core.UnixCompatibility;
+
+public static class SteamCompatibilityTool
+{
+    public static bool IsSteamInstalled => Directory.Exists(Program.Config.SteamPath);
+
+    public static bool IsSteamFlatpakInstalled => Directory.Exists(Program.Config.SteamFlatpakPath);
+
+    public static bool IsSteamToolInstalled => Directory.Exists(Path.Combine(Program.Config.SteamPath, "compatibilitytools.d", "xlcore"));
+
+    public static bool IsSteamFlatpakToolInstalled => Directory.Exists(Path.Combine(Program.Config.SteamFlatpakPath, "compatibilitytools.d", "xlcore"));
+
+    private static string findXIVLauncherFiles()
+    {
+        return System.AppDomain.CurrentDomain.BaseDirectory;
+    }
+
+    public static void CreateTool(string path)
+    {
+        var compatfolder = new DirectoryInfo(Path.Combine(path, "compatibilitytools.d"));
+        compatfolder.Create();
+        var destination = new DirectoryInfo(Path.Combine(compatfolder.FullName, "xlcore"));
+        if (File.Exists(destination.FullName))
+            File.Delete(destination.FullName);
+        if (destination.Exists)
+            destination.Delete(true);
+        
+        destination.Create();
+        destination.CreateSubdirectory("XIVLauncher");
+        destination.CreateSubdirectory("bin");
+        destination.CreateSubdirectory("lib");
+
+        var xlcore = new FileInfo(Path.Combine(destination.FullName, "xlcore"));
+        var compatibilitytool_vdf = new FileInfo(Path.Combine(destination.FullName, "compatibilitytool.vdf"));
+        var toolmanifest_vdf = new FileInfo(Path.Combine(destination.FullName, "toolmanifest.vdf"));
+        var openssl_fix = new FileInfo(Path.Combine(destination.FullName, "openssl_fix.cnf"));
+
+        xlcore.Delete();
+        compatibilitytool_vdf.Delete();
+        toolmanifest_vdf.Delete();
+        openssl_fix.Delete();
+
+        using (var fs = xlcore.Create())
+        {
+            var resource = Assembly.GetExecutingAssembly().GetManifestResourceStream("xlcore");
+            resource.CopyTo(fs);
+            fs.Close();
+        }
+
+        // File.SetUnixFileMode() doesn't exist, for some reason, so just run chmod
+        var psi = new ProcessStartInfo("/bin/chmod");
```

Comment:
Very strange that SetUnixFileMode isn't available here.

Reply:
It's not available in dotnet 6. We can switch to it if we upgrade to 8. https://learn.microsoft.com/en-us/dotnet/api/system.io.file.setunixfilemode?view=net-8.0. You can see in the Applies To section that it's for 7 and later.